spring-batch-core/src/test/java/org/springframework/batch/core/job/builder/FlowJobBuilderTests.java

```diff
@@ -419,4 +424,38 @@ public JdbcTransactionManager transactionManager(DataSource dataSource) {
 
 	}
 
+	@Test
+	public void testBuildSplitWithParallelFlow() throws InterruptedException {
+		CountDownLatch countDownLatch = new CountDownLatch(1);
+		Step longExecutingStep = new StepBuilder("longExecutingStep", jobRepository).tasklet((stepContribution, b) -> {
+			Thread.sleep(500L);
+			return RepeatStatus.FINISHED;
+		}, new ResourcelessTransactionManager()).build();
+
+		Step interruptedStep = new StepBuilder("interruptedStep", jobRepository).tasklet((stepContribution, b) -> {
+			stepContribution.getStepExecution().setTerminateOnly();
+			return RepeatStatus.FINISHED;
+		}, new ResourcelessTransactionManager()).build();
+
+		Step nonExecutableStep = new StepBuilder("nonExecutableStep", jobRepository).tasklet((stepContribution, b) -> {
+			countDownLatch.countDown();
+			return RepeatStatus.FINISHED;
+		}, new ResourcelessTransactionManager()).build();
+
+		Flow twoStepFlow = new FlowBuilder<SimpleFlow>("twoStepFlow").start(longExecutingStep)
+			.next(nonExecutableStep)
+			.build();
+		Flow interruptedFlow = new FlowBuilder<SimpleFlow>("interruptedFlow").start(interruptedStep).build();
+
+		Flow splitFlow = new FlowBuilder<Flow>("splitFlow").split(new SimpleAsyncTaskExecutor())
+			.add(interruptedFlow, twoStepFlow)
+			.build();
+		FlowJobBuilder jobBuilder = new JobBuilder("job", jobRepository).start(splitFlow).build();
+		jobBuilder.preventRestart().build().execute(execution);
+
+		boolean isExecutedNonExecutableStep = countDownLatch.await(1, TimeUnit.SECONDS);
+		assertEquals(BatchStatus.STOPPED, execution.getStatus());
```

Comment:
Finished with `BatchStatus.STOPPED` without executing `nonExecutableStep`.
This test would fail with executing `nonExecutableStep` if it runs on the current branch.